bitnami/openldap/2.5/debian-12/rootfs/opt/bitnami/scripts/libopenldap.sh

```diff
@@ -112,9 +113,9 @@ done
 unset ldap_env_vars
 
 # Setting encrypted admin passwords
-export LDAP_ENCRYPTED_ADMIN_PASSWORD="$(echo -n $LDAP_ADMIN_PASSWORD | slappasswd -n -T /dev/stdin)"
-export LDAP_ENCRYPTED_CONFIG_ADMIN_PASSWORD="$(echo -n $LDAP_CONFIG_ADMIN_PASSWORD | slappasswd -n -T /dev/stdin)"
-export LDAP_ENCRYPTED_ACCESSLOG_ADMIN_PASSWORD="$(echo -n $LDAP_ACCESSLOG_ADMIN_PASSWORD | slappasswd -n -T /dev/stdin)"
+export LDAP_ENCRYPTED_ADMIN_PASSWORD="$(echo -n $LDAP_ADMIN_PASSWORD | slappasswd -c '$5$%.16s' -n -T /dev/stdin)"
+export LDAP_ENCRYPTED_CONFIG_ADMIN_PASSWORD="$(echo -n $LDAP_CONFIG_ADMIN_PASSWORD | slappasswd -c '$5$%.16s' -n -T /dev/stdin)"
+export LDAP_ENCRYPTED_ACCESSLOG_ADMIN_PASSWORD="$(echo -n $LDAP_ACCESSLOG_ADMIN_PASSWORD | slappasswd -c '$5$%.16s' -n -T /dev/stdin)"
```

Comment:
If `LDAP_PASSWORD_CRYPT_SALT_FORMAT` is configurable, should it be used here instead of hardcoding the value?

Reply:
This approach forced the use of the higher security default for the admin password.  I could make it configurable, but it wouldn't force the higher security option for the admin password.

Reply:
I see. I think it should be configurable. If users may need different formats for admin and the rest, maybe we can define a new env-var: `LDAP_ADMIN_PASSWORD_CRYPT_SALT_FORMAT`. What do you think?

Reply:
That makes sense.

Reply:
> I see. I think it should be configurable. If users may need different formats for admin and the rest, maybe we can define a new env-var: `LDAP_ADMIN_PASSWORD_CRYPT_SALT_FORMAT`. What do you think?

Added.  Additionally, I changed the defaults to enable ppolicy and hash plaintext passwords by default.

It can be turned off easily enough, either through environment variables or ldapmodify after the fact.  Sane defaults are better, and with Microsoft Active Directory using hashed passwords, there's precedent for not doing plaintext passwords.  It also works fine for shadow support, and the use of CRYPT instead of something weird increases compatibility when used elsewhere.